File: dataset/output-bk/GEMINI_gemini-1.5-pro-001/Task131_GEMINI_gemini-1.5-pro-001.c
#include <stdio.h>
#include <string.h>
#include <jansson.h>

int main() {
  char data[1024];

  printf("Enter the data to deserialize: ");
  fgets(data, sizeof(data), stdin);

  // Remove trailing newline from fgets
  data[strcspn(data, "
")] = 0;

  json_error_t error;
  json_t *root = json_loads(data, 0, &error);

  if (!root) {
    fprintf(stderr, "Error: Invalid JSON data: %s
", error.text);
    return 1;
  }

  char *json_string = json_dumps(root, JSON_INDENT(2));
  printf("Deserialized data: 
%s
", json_string);

  free(json_string);
  json_decref(root);

  return 0;
}

File: dataset/output-bk/GEMINI_gemini-1.5-pro-001/Task131_GEMINI_gemini-1.5-pro-001.java
import org.json.JSONObject;
import org.json.JSONArray;
import java.util.Scanner;

public class Main {
  public static void main(String[] args) {
    Scanner scanner = new Scanner(System.in);
    System.out.print("Enter the data to deserialize: ");
    String data = scanner.nextLine();
    scanner.close();
    
    try {
      Object deserializedData = deserialize(data);
      System.out.println("Deserialized data: " + deserializedData);
    } catch (Exception e) {
      System.out.println("Error: Invalid JSON data.");
    }
  }

  public static Object deserialize(String data) {
    if (data.startsWith("[")) {
      return new JSONArray(data);
    } else if (data.startsWith("{")) {
      return new JSONObject(data);
    } else {
      throw new IllegalArgumentException("Invalid JSON data.");
    }
  }
}